InformationScripting/src/queries/CompositeQuery.cpp

```diff
@@ -76,7 +77,13 @@ QList<TupleSet> CompositeQuery::execute(QList<TupleSet> input)
 					return m.outputFrom_ == currentNode && m.outputIndex_ == outIndex;
 				});
 				Q_ASSERT(inputIt != receiver->inputMap_.end());
-				auto output = hasOutput ? currentNode->calculatedOutputs_[outIndex] : TupleSet();
+				TupleSet output;
+				if (hasOutput)
+				{
+					// early abort in case of error:
+					if (currentNode->calculatedOutputs_[outIndex]) output = currentNode->calculatedOutputs_[outIndex].value();
+					else return { currentNode->calculatedOutputs_[outIndex].error() };
```

Comment:
Hmm, now that I think about it, I'm not sure about the merge function and where it should be. We want to transition from a `QList<Optional>` to `Optional<QList<`. See also some of my comments below about this. Perhaps, logically, the correct thing to have is indeed `Optional<QList<TupleSet>>` as the return type of `Query::execute`.